skpro/regression/gradient_boosting/_ngboost.py

```diff
@@ -0,0 +1,337 @@
+            natural_gradient=True,
+            n_estimators=self.n_estimators,
+            learning_rate=self.learning_rate,
+            minibatch_frac=self.minibatch_frac,
+            col_sample=self.col_sample,
+            verbose=self.verbose,
+            verbose_eval=self.verbose_eval,
+            tol=self.tol,
+            random_state=self.random_state,
+            validation_fraction=self.validation_fraction,
+            early_stopping_rounds=self.early_stopping_rounds,
+        )
+        self.ngb.fit(X, y)
+        self._is_fitted = True
+        return self
+
+    def _predict(self, X):
+        return self.ngb.predict(X)
+
+    def _pred_dist(self, X):
+        return self.ngb.pred_dist(X)
+
+    def _ngb_dist_to_skpro(self, pred_mean, pred_std, index, columns):
+        """
+        Convert NGBoost distribution object to skpro BaseDistribution object.
+
+        Parameters
+        ----------
+        pred_mean, pred_std and index and columns.
+
+        Returns
+        -------
+        skpro_dist (skpro.distributions.BaseDistribution):
+        Converted skpro distribution object.
+        """
+        skpro_dist = None
+        if self.dist == "Normal":
+            from skpro.distributions.normal import Normal
+
+            skpro_dist = Normal(
+                mu=pred_mean, sigma=pred_std, index=index, columns=columns
+            )
+
+        if self.dist == "Laplace":
+            from skpro.distributions.laplace import Laplace
+
+            skpro_dist = Laplace(
+                mu=pred_mean, scale=pred_std, index=index, columns=columns
+            )
+
+        if self.dist == "TDistribution":
+            from skpro.distributions.t import TDistribution
+
+            skpro_dist = TDistribution(
+                mu=pred_mean, sigma=pred_std, index=index, columns=columns
+            )
+
+        if self.dist == "Poisson":
+            from skpro.distributions.poisson import Poisson
+
+            skpro_dist = Poisson(mu=pred_mean, index=index, columns=columns)
+
+        if self.dist == "LogNormal":
+            from skpro.distributions.lognormal import LogNormal
+
+            skpro_dist = LogNormal(
+                mu=pred_mean, sigma=pred_std, index=index, columns=columns
+            )
+
+        return skpro_dist
+
+    def _predict_proba(self, X):
+        """Predict distribution over labels for data from features.
+
+        State required:
+            Requires state to be "fitted".
+
+        Accesses in self:
+            Fitted model attributes ending in "_"
+
+        Parameters
+        ----------
+        X : pandas DataFrame, must have same columns as X in `fit`
+            data to predict labels for
+
+        Returns
+        -------
+        y : skpro BaseDistribution, same length as `X`
+            labels predicted for `X`
+        """
+        X = self._check_X(X)
+
+        # Convert NGBoost distribution to skpro BaseDistribution
+        pred_mean = self._predict(X=X)
```

Comment:
The general idea makes sense, though I would do it differently:

1. just call `pred_dist` from `ngboost`. This is a full parameteric distirbution.
2. in the `_ngb_dist_to_skpro`, extract the parameters from the `ngboost` distribution and plug them into the `skpro` distribution.

No need to take detours via the mean or `_predict`, especially since some of the distributions are not directly parameterized in terms of the mean.

Reply:
I have now taken care as to what the value actually returns from the `ngboost` and converted it to the respective value in `skpro`. It has also been explained using the comments as to what the `distribution` is and what the `parameters` indicate.